pandas/__init__.py

```diff
@@ -202,8 +202,29 @@
         FutureWarning,
         stacklevel=2,
     )
-# Don't allow users to use pandas.os or pandas.warnings
-del os, warnings
+# Don't allow users to use pandas.os
+del os
+
+# DeprecationWarning for missing pyarrow
+from pandas.compat.pyarrow import pa_version_under10p1
+
+if pa_version_under10p1:
+    import warnings
+    from pandas.compat._optional import VERSIONS
```

Comment:
This will make `pandas.VERSIONS` available when PyArrow is not installed. Do we want to do a `del VERSIONS` inside the `if`?

Reply:
Wow - didn't know that happened.

Thanks for the catch.